### DISCUSSION TITLE
vue3 ts项目如何使用onShow

### DISCUSSION THREAD

**Discussion by holyrong** [NervJS/taro#11082] [Q & A] [answered]
项目使用：vue3 typescript  
页面：setup <script lang="ts" setup>
**如何使用Taro的onShow事件函数？**

**Comment by Chen-jj** [comment, marked as answer]
等等 3.4 版本:https://docs.taro.zone/docs/composition-api

**Reply by holyrong** [reply]
> 
谢谢回复！
大概到什么时候可用？
目前在写小程序，从uni-app迁移过来。发现Taro  NutUI不能写onShow，现在一些在onShow的逻辑不知道怎么处理。

**Reply by Hilery** [reply]
js可以这样处理 ts不太会用。
<script>
  import { eventCenter, getCurrentInstance } from '@tarojs/taro'

  export default {
    mounted () {
      eventCenter.once(getCurrentInstance().router.onShow, () => {
        console.log('onShow')
      })
    }
  }
</script>

**Comment by Chen-jj** [comment]
今天发布的

**Comment by holyrong** [comment]
![image](https://user-images.githubusercontent.com/8318475/149084225-9a256a5d-8ca5-4c76-a3b8-e9f7563a6c91.png)
